packages/opentelemetry-exporter-collector-grpc/test/CollectorExporterNodeBase.test.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { collectorTypes } from '@opentelemetry/exporter-collector';
+import { ReadableSpan } from '@opentelemetry/sdk-trace-base';
+
+import * as assert from 'assert';
+import { CollectorExporterNodeBase } from '../src/CollectorExporterNodeBase';
+import { CollectorExporterConfigNode, ServiceClientType } from '../src/types';
+import { mockedReadableSpan } from './helper';
+
+class MockCollectorExporter extends CollectorExporterNodeBase<
+  ReadableSpan,
+  ReadableSpan[]
+> {
+  /**
+   * Callbacks passed to _send()
+   */
+  sendCallbacks: {
+    onSuccess: () => void;
+    onError: (error: collectorTypes.CollectorExporterError) => void;
+  }[] = [];
+
+  getDefaultUrl(config: CollectorExporterConfigNode): string {
+    return '';
+  }
+
+  getDefaultServiceName(config: CollectorExporterConfigNode): string {
+    return '';
+  }
+
+  convert(spans: ReadableSpan[]): ReadableSpan[] {
+    return spans;
+  }
+
+  getServiceClientType() {
+    return ServiceClientType.SPANS;
+  }
+
+  getServiceProtoPath(): string {
+    return 'opentelemetry/proto/collector/trace/v1/trace_service.proto';
+  }
+}
+
+// Mocked _send which just saves the callbacks for later
+MockCollectorExporter.prototype['_send'] = function _sendMock(
```

Comment:
Is square bracket notation used here to bypass eslint's convention not to access `_` prefixed properties?

Reply:
it's actually to access private fields. Something we allow in tests